src/discover/discover.ml

```diff
@@ -11,13 +11,20 @@ let os c =
     | _ -> Other
 
 let from_pkg_config c =
-  let fallback = ([], [ "-lX11" ]) in
+  let fallback = ([], [ "-lX11"; "-lXft" ]) in
   match C.Pkg_config.get c with
   | None -> fallback
-  | Some pc -> (
-      match C.Pkg_config.query pc ~package:"x11" with
-      | None -> fallback
-      | Some { cflags; libs } -> (cflags, libs))
+  | Some pc ->
+      let packages = [("x11","-lX11"); ("xft", "-lXft")] in
+      let cflags, libs =
+      packages
+      |> List.map (fun (package, fallback) ->
+          match C.Pkg_config.query pc ~package with
+          | None -> [], [fallback]
+          | Some { cflags; libs } -> cflags, libs)
+      |> List.split
+      in
+      List.concat cflags, List.concat libs
```

Comment:
Ouch.  `pkg-query` handles multiple package names just fine, e.g. `pkg-config --libs xft x11` gives `-lXft -lX11`, no need for multiple invocations.  But of cours Dune configurator doesn't support this.  

The whole `discover.ml` file is an embarrassment.  It would be 3 times shorter as a shell script.

Also: it would be prudent to link `-lXft` before `-lX11`, as the former certainly uses symbols from the latter.